src/psyclone/tests/psyGen_test.py

```diff
@@ -4511,17 +4657,59 @@ def test_fparser2astprocessor_handling_binaryopbase(f2008_parser):
     from fparser.common.readfortran import FortranStringReader
     from fparser.two.Fortran2003 import Execution_Part
     reader = FortranStringReader("x=1+4")
-    fparser2binary_operation = Execution_Part.match(reader)[0][0].items[2]
+    fp2binaryop = Execution_Part.match(reader)[0][0].items[2]
 
     fake_parent = Node()
     processor = Fparser2ASTProcessor()
-    processor.process_nodes(fake_parent, [fparser2binary_operation], None)
+    processor.process_nodes(fake_parent, [fp2binaryop], None)
     # Check a new node was generated and connected to parent
     assert len(fake_parent.children) == 1
     new_node = fake_parent.children[0]
     assert isinstance(new_node, BinaryOperation)
     assert len(new_node.children) == 2
-    assert new_node._operator == '+'
+    assert new_node._operator == BinaryOperation.Operator.ADD
+
+    # Test parsing all supported binary operators.
+    testlist = (('+', BinaryOperation.Operator.ADD),
+                ('-', BinaryOperation.Operator.SUB),
+                ('*', BinaryOperation.Operator.MUL),
+                ('/', BinaryOperation.Operator.DIV),
+                ('**', BinaryOperation.Operator.POW),
+                ('==', BinaryOperation.Operator.EQ),
+                ('.eq.', BinaryOperation.Operator.EQ),
+                ('.EQ.', BinaryOperation.Operator.EQ),
+                ('/=', BinaryOperation.Operator.NE),
+                ('.ne.', BinaryOperation.Operator.NE),
+                ('>', BinaryOperation.Operator.GT),
+                ('.GT.', BinaryOperation.Operator.GT),
+                ('<', BinaryOperation.Operator.LT),
+                ('.lt.', BinaryOperation.Operator.LT),
+                ('>=', BinaryOperation.Operator.GE),
+                ('.ge.', BinaryOperation.Operator.GE),
+                ('<=', BinaryOperation.Operator.LE),
+                ('.LE.', BinaryOperation.Operator.LE),
+                ('.and.', BinaryOperation.Operator.AND),
+                ('.or.', BinaryOperation.Operator.OR),
+                )
+
+    for opstring, expected in testlist:
+        fake_parent = Node()
+        fp2binaryop.items = (fp2binaryop.items[0], opstring,
```

Comment:
Please add comment describing what you're doing, e.g. "Manipulate the fparser2 ParseTree so that it contains the operator under test"

Reply:
Added comment